framework/test/mocha/functional/http/get/transactions.js

```diff
@@ -357,6 +357,111 @@ describe('GET /api/transactions', () => {
 						});
 					});
 			});
+
+			describe('asset field', () => {
+				it('using type 0 should return asset field with correct properties', async () => {
+					return transactionsEndpoint
+						.makeRequest({ type: transactionTypes.SEND }, 200)
+						.then(res => {
+							expect(res.body.data).to.not.empty;
+
+							res.body.data.map(transaction =>
+								expect(Object.keys(transaction.asset).length).to.be.below(2)
+							);
+						});
+				});
+
+				it('using type 1 should return asset field with correct properties', async () => {
+					return transactionsEndpoint
+						.makeRequest({ type: transactionTypes.SIGNATURE }, 200)
+						.then(res => {
+							expect(res.body.data).to.not.empty;
+
+							res.body.data.map(transaction => {
+								expect(transaction.asset).to.have.property('signature');
+								expect(transaction.asset.signature).to.have.property(
+									'publicKey'
+								);
+								return expect(transaction.asset.signature.publicKey).to.be.a(
+									'string'
+								);
+							});
+						});
+				});
+
+				it('using type 2 should return asset field with correct properties', async () => {
+					return transactionsEndpoint
+						.makeRequest({ type: transactionTypes.DELEGATE }, 200)
+						.then(res => {
+							expect(res.body.data).to.not.empty;
+
+							res.body.data.map(transaction => {
+								expect(transaction.asset.delegate).to.have.property(
+									'publicKey'
+								);
+								expect(transaction.asset.delegate).to.have.property('username');
+								return expect(transaction.asset.delegate).to.have.property(
+									'address'
+								);
+							});
+						});
+				});
+
+				it('using type 3 should return asset field with correct properties', async () => {
+					return transactionsEndpoint
+						.makeRequest({ type: transactionTypes.VOTE }, 200)
+						.then(res => {
+							expect(res.body.data).to.not.empty;
+
+							res.body.data.map(transaction => {
+								expect(Object.keys(transaction.asset).length).to.equal(1);
+								return expect(transaction.asset.votes).to.be.an('array');
```

Comment:
Please validate the array length - up to 33 votes

Reply:
@MaciejBaj this does not work as the transaction from `Genesis_block.json` holds a tx with 101 votes although limit is 33. What about this?

Reply:
Filter out Genesis transaction by account.